File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        // A: it will gives head of node after swapping all adj nodes
        
        if(head == NULL || head->next == NULL)  return head;
        
        ListNode* temp = NULL;
        ListNode* temp2 = head;
        ListNode* head2 = head;
        int i = 0;
        while(i<2){
            head2 = head2->next;
            temp2->next = temp;
            temp = temp2;
            temp2 = head2;
            i+=1;
        }
        
        head->next = swapPairs(head2);
        
        return temp;
        
        
    }
};

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if(head == None or head.next == None):
            return head
        
        curr = head
        prev = None
        
        while(curr and curr.next):
            temp = curr.next
            curr.next = temp.next
            if(prev == None):
                head = temp
            else:
                prev.next = temp
            temp.next = curr
            
            prev = curr            
            curr = curr.next
        
        return head
            
            